source/ThemeManager.cpp

```diff
@@ -0,0 +1,31 @@
+/*
+** Project FileShare, 2022
+**
+** Author Léo Lhuile
+**
+** Started on  Thu Jul 24 15:44:13 2025 Léo Lhuile
+** Last update Thu Jul 24 15:44:13 2025 Léo Lhuile
+**
+** ThemeManager.cpp : Implementation of ThemeManager class
+*/
+
+#include "ThemeManager.hpp"
+
+namespace FileShare::GUI {
+    ThemeManager::ThemeManager()
+    {
+        this->setTheme(this->currentTheme);
+    }
+
+    ThemeManager &ThemeManager::getInstance()
+    {
+        static ThemeManager instance;
+        return instance;
+    }
+
+    void ThemeManager::setTheme(const std::string &theme)
+    {
```

Comment:
project style is to have opening braces on same line :stuck_out_tongue: 
```suggestion
    void ThemeManager::setTheme(const std::string &theme) {
```